Update payload.go

vanugrah opened the pull request:
The proposed change was missed in this PR: 
https://github.com/grafana/agent/pull/5654

<!--

CONTRIBUTORS GUIDE: https://github.com/grafana/agent/blob/main/docs/developer/contributing.md#updating-the-changelog

If this is your first PR or you have not contributed in a while, we recommend
taking the time to review the guide. It gives helpful instructions for
contributors around things like how to update the changelog.

-->

#### PR Description

#### Which issue(s) this PR fixes

<!-- Uncomment the following line if you want that GitHub issue gets automatically closed after merging the PR -->
<!-- Fixes #issue_id -->

#### Notes to the Reviewer

#### PR Checklist

<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [ ] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

tpaschalis commented:
It looks OK to add this since the measurement does include the type in the matching faro.receiver component.

https://github.com/grafana/agent/blob/287b1c68cae77768d70ff20f7d753db943586393/component/faro/receiver/internal/payload/payload.go#L228

@vanugrah Could you please also add a CHANGELOG entry under Enhancements so you can receive the credit due? 🙌

rlankfo reviewed (changes requested):
Looks good to me once the tests are updated. 👍

vanugrah commented:
Hey folks - thanks so much for your review. I have updated the failing test and added a changelog entry. Please let me know if any further changes are required. I think the CI run has has to be approved by an admin.

rlankfo reviewed (approved):
LGTM
